applications/FluidDynamicsApplication/custom_utilities/embedded_discontinuous_data.h

```diff
@@ -128,7 +140,27 @@ static int Check(
  */
 bool IsCut()
 {
-    return (NumPositiveNodes > 0) && (NumNegativeNodes > 0);
+    if (IsAusasIncised()) {
+        return false;
```

Comment:
I added this check to ensure that `IsCut()` returns the correct value.
Correct treatment is necessary, because for an incised element elemental distances of the "original" and (!) extrapolated intersections will be saved to `ELEMENTAL_DISTANCES`, if the flag `CALCULATE_EDGE_DISTANCES_EXTRAPOLATED` is given to `CalculateDiscontinuousDistanceToSkinProcess`.
Alternatively, it would be sufficient, and maybe faster, but less, clear to always call `IsAusasIncised()` before (!) calling `IsCut()` in the element.

Reply:
Third one makes sense, but we can leave it for a future small PR. For the moment, I'd stick to 2, as it is a robust solution with minimal overhead.